src/enzo/ActiveParticle_SmartStar.C

```diff
@@ -0,0 +1,1058 @@
+	centralpos[0] = thisGrid->CellLeftEdge[0][i] + 0.5*thisGrid->CellWidth[0][i];
+	centralpos[1] = thisGrid->CellLeftEdge[1][j] + 0.5*thisGrid->CellWidth[1][j];
+	centralpos[2] = thisGrid->CellLeftEdge[2][k] + 0.5*thisGrid->CellWidth[2][k];
+	
+#if COOLING_TIME
+	// 4. t_cool < t_freefall (skip if T > 11000 K)
+	dtot = ( density[index] + data.DarkMatterDensity[index] ) * 
+	  data.DensityUnits;
+	tdyn = sqrt(3.0 * M_PI / 32.0 / GravConst / dtot) / data.TimeUnits;
+	if (tdyn < data.CoolingTime[index] && 
+	    data.Temperature[index] > 1.1e4)
+	  continue;
+#endif
+#if MASSTHRESHOLDCHECK	
+	TotalMass = thisGrid->FindMassinGrid(data.DensNum);
+	/* Mass Threshold check */
+	/* The control region should contain a mass greater than the mass threshold */
+	if(TotalMass*ConverttoSolar < (double)MASSTHRESHOLD) {	  
+	  continue;
+	}
+#if SSDEBUG_TOTALMASS
+	printf("%s: Total Mass in Accretion Region = %g Msolar (Threshold = %g)\n", __FUNCTION__,
+	       TotalMass*ConverttoSolar, (double)MASSTHRESHOLD);
+#endif
+#endif
+#if MINIMUMPOTENTIAL
+#if CALCDIRECTPOTENTIAL
+	if(PotentialField == NULL) {
+	  PotentialField = new float[size];
+	  thisGrid->CalculatePotentialField(PotentialField, data.DensNum, data.DensityUnits, data.TimeUnits,data.LengthUnits);
+	}
+#endif
+	/* 4. Gravitational Minimum Check */
+	/* 
+	 * Find the minimum of the potential over a Jeans Length. 
+	 */
+	double JLength = JeansLength(CellTemperature, density[index],
+				     data.DensityUnits)/data.LengthUnits;
+	GravitationalMinimum  = thisGrid->FindMinimumPotential(centralpos, JLength*64.0,
+							       PotentialField);
+	if(PotentialField[index] > GravitationalMinimum) {
+#if SSDEBUG
+	  printf("FAILURE: GravitationalMinimum = %g\t " \
+	    "PotentialField[index] = %g\n\n", GravitationalMinimum, PotentialField[index]);
+#endif
+	  continue;
+	}
+
+
+#endif
+#ifdef GRAVENERGY
+	static int mincount = 0;
+	/* 5. Jeans Instability Check */
+	// printf("Pos = (%f, %f, %f)\n", centralpos[0], centralpos[1], centralpos[2]);
+	// printf("GENum = %d\t GravPotentialNum = %d\t DensNum = %d\n", data.GENum,
+	//        data.GravPotentialNum, data.DensNum);
+	// printf("Total Energy = %1.20g\n", thisGrid->BaryonField[data.TENum][index]);
+	// printf("Thermal Energy = %1.20g\n", thisGrid->BaryonField[data.GENum][index]);
+	// printf("Total - Thermal = %g\n", thisGrid->BaryonField[data.TENum][index] -
+	//        thisGrid->BaryonField[data.GENum][index]);
+	// float vel = pow(velx[index], 2.0) + pow(vely[index], 2.0) + pow(velz[index], 2.0);
+	// printf("Kinetic Energy =  %1.20g\n", 0.5*vel);
+	// printf("Total - Thermal - Kinetic = %g\n", thisGrid->BaryonField[data.TENum][index] -
+	//        thisGrid->BaryonField[data.GENum][index] - 0.5*vel);
+	// printf("Gravitational Energy = %1.20g\n", thisGrid->BaryonField[data.GravPotentialNum][index]);
+	// printf("Potential Field = %1.20g\n", thisGrid->PotentialField[index]);
+	// printf("mass = %g\n", mass);
```

Comment:
Remove these commented out bunch of print statements.

Reply:
Done